### ISSUE TITLE
Clean up ioctl path

### ISSUE THREAD

**Issue by luqmana** [oxidecomputer/opte#333]
Ioctls originally went through a DLD specific mechanism but after #321 we'll be handling them directly. As part of that change the general structure wasn't changed to minimize the changes needed. This issue is reevaluate and see if we should handle things differently now that we don't have to stick to the dld ioc semantics. (We may also decide it's ok as-is but nice to have an issue to track this).

**Comment by FelixMcFelix** [comment]
Expanding on this a bit with some thoughts I had while working on #529:

* `OpteCmd` mixes ioctls intended for XDE/OPTE ports in general, ioctls which ask for state common to all `NetworkImpl`s (parametrised on `Impl`-specific types), and ioctls specific to a given port type.
  * List ports, list layers, clear UFTs/LFTs, etc. all fit into the first category.
  * Dump UFT/Layer/... fit into the second -- these are common aspects of all OPTE ports, but each entry contains `Impl`-specific types. In the limit, different `NetworkImpl`s could have different Flow key types, and per-flow/entry metadata.
  * Management of rules depends entirely on cmds exclusively using `oxide-vpc` functionality. Obviously we only have this dataplane design today.
* Callers on each side of the ioctl divide need to manually specify the response type expected from a given request. This should be statically known.
* We have many methods with code of the form `let req = XyzReq { ... }; let cmd = OpteCmd::Xyz; let resp = run_cmd_ioctl(self.device.as_raw_fd(), cmd, Some(&req))?;`. `cmd` should be trivially derived from `req`.
* We still don't have a version number on `oxide-vpc`, and this should be served/verified in-band somehow.

This ties into later asks as well -- for #446 we'll need to make rules and finite resources serialisable and deserialisable dependent on the type of port being targeted.

This isn't a priority and is not at all a prerequisite for flow stats and metadata, but I found it worth it to get this down while it's still fresh.
